### ISSUE TITLE
Enums: keep original value option

### ISSUE THREAD

**Issue by PseudoSky** [IBM/openapi-to-graphql#357]
**Describe the bug**
There isn't a way to keep the literal enum values from the oas spec.
The external api may require the enum to be exactly the same as what is defined.
Regardless of the options the values always seem to be sanitized to uppercase + underscore.

**Proposal**

- Map the enum values to valid values
- Store the mapping in state somehow
- Reverse the mapping of the values before the request is made

**OR**

I think it would be much more useful to add a fetch override option to the configuration -
so the consumer can implement more complex solutions / fix assumptions made by the library

- More easily map custom resolvers
- Add delegated resolvers
- Linking is currently near impossible with message envelopes
- Many more

No idea how complicated this would be within the library, I'm trying to avoid attempting it
Looking for thoughts on this

**To Reproduce**
Example
```yaml
TextFilter:
      description: Text filter
      required:
        - key
        - operator
        - value
      properties:
        key:
          type: string
          description: Key on an object
          enum:
            - id
            - name
            - assetType
        operator:
          type: string
          enum:
            - Equals
            - Does not equal
            - Contains
            - Does not contain
```

**Expected behavior**

I need the value posted to the external api to be "Does not equal" rather than "DOES_NOT_EQUAL"

**Comment by Alan-Cha** [comment]
@PseudoSky Thanks for the issue and sorry for the late response. Could you verify that, indeed, the sanitzation/desanitization is not working? OtG is supposed to sanitize enum values to ALL_CAPS (to reflect GraphQL conventions), and desanitize back to the original values when the API call is made. I just want to make sure that this is not a misunderstanding.

If indeed, desanitization is functioning incorrectly, then this could potentially related to #274 where I discuss our naive approach to sanitization/desanitization mapping. However, if that is the case, I would assume that we would still see some kind of desanitization going on, just to an incorrect value.

However, I do agree, there probably should be an option that allows you to control whether the values are sanitized or not. I believe there are some strict limitations on the characters you can use in enum values but besides that, we can try to preserve as much of the original value as possible. 

***

> More easily map custom resolvers

We do offer custom resolvers via the `customResolvers` option. You can find that option [here](https://github.com/IBM/openapi-to-graphql/tree/master/packages/openapi-to-graphql#options). However, the enum values will be sanitized. If I recall correctly, we also expose the `sanitize()` and `desanitize()` functions. 

> Add delegated resolvers

To be honest, I am not very familiar with schema delegation but it seems to me that there isn't a clean, straightforward way to do this programmatically without some external configuration and at that point, `customResolvers` is probably the best way forward.

**Comment by nlundquist** [comment]
additionally this is undesirable behavior for some. for example the GeoJSON standard specifies that enum names must be in titlecase, eg `type: "Point"`. I want my generated GraphQL schema to permit the query and input of valid GeoJSON types that must have this casing regardless of the GQL norms.

**Comment by Alan-Cha** [comment]
This should be resolved via #365
